core/Test/Tasty/Core.hs

```diff
@@ -398,6 +432,17 @@ trivialFold = TreeFold
   , foldAfter = \_ _ _ b -> b
   }
 
+
+-- | Indicates whether a test matched in an evaluated subtree. If no filter was
+-- used, tests always match.
+type TestMatched = Any
+
+-- | Used to force tests to be included, even if they would be filtered out by
+-- a user's filter. This is used to force dependencies of a test to run. For
+-- example, if test @A@ depends on test @B@ and test @A@ is selected to run, test
+-- @B@ will be forced to match.
```

Comment:
Does this apply universally or only to sequential test groups?

Reply:
Only sequential test groups, I've extended the documentation to highlight this.